server/src/postgis/best_path.rs

```diff
@@ -0,0 +1,392 @@
+                    .map_err(|_| PathError::InvalidStartNode)?;
+            }
+            _ => {
+                postgis_error!(
+                    "(try_from BestPathRequest) invalid start node type: {:?}",
+                    request.start_type
+                );
+                return Err(PathError::InvalidStartNode);
+            }
+        }
+
+        let node_start_id = request.node_start_id;
+        let node_uuid_end = match uuid::Uuid::parse_str(&request.node_uuid_end) {
+            Ok(uuid) => uuid,
+            Err(_) => return Err(PathError::InvalidEndNode),
+        };
+
+        let time_start: DateTime<Utc> = match request.time_start {
+            None => chrono::Utc::now(),
+            Some(time) => time.into(),
+        };
+
+        let time_end: DateTime<Utc> = match request.time_end {
+            None => chrono::Utc::now() + chrono::Duration::days(1),
+            Some(time) => time.into(),
+        };
+
+        if time_end < time_start {
+            return Err(PathError::InvalidTimeWindow);
+        }
+
+        if time_end < Utc::now() {
+            return Err(PathError::InvalidEndTime);
+        }
+
+        Ok(PathRequest {
+            node_start_id,
+            node_uuid_end,
+            time_start,
+            time_end,
+        })
+    }
+}
+
+/// Get the best path from a vertiport to another vertiport
+async fn best_path_vertiport_source(
+    stmt: tokio_postgres::Statement,
+    client: deadpool_postgres::Client,
+    request: PathRequest,
+) -> Result<Vec<tokio_postgres::Row>, PathError> {
+    let Ok(node_start_id) = Uuid::parse_str(&request.node_start_id) else {
+        postgis_error!("(best_path_vertiport_source) could not parse start node id into UUID: {}", request.node_start_id);
+        return Err(PathError::InvalidStartNode);
+    };
+
+    match client
+        .query(
+            &stmt,
+            &[
+                &node_start_id,
+                &request.node_uuid_end,
+                &request.time_start,
+                &request.time_end,
+            ],
+        )
+        .await
+    {
+        Ok(results) => Ok(results),
+        Err(e) => {
+            println!(
+                "(best_path_vertiport_source) could not request routes: {}",
+                e
+            );
+            Err(PathError::Unknown)
+        }
+    }
+}
+
+/// Get the best path from a aircraft to another vertiport
+async fn best_path_aircraft_source(
+    stmt: tokio_postgres::Statement,
+    client: deadpool_postgres::Client,
+    request: PathRequest,
+) -> Result<Vec<tokio_postgres::Row>, PathError> {
+    match client
+        .query(
+            &stmt,
+            &[
+                &request.node_start_id,
+                &request.node_uuid_end,
+                &request.time_start,
+                &request.time_end,
+            ],
+        )
+        .await
+    {
+        Ok(results) => Ok(results),
+        Err(e) => {
+            println!(
```

Comment:
Should be: `postgis_error!(`

Reply:
https://github.com/Arrow-air/svc-gis/commit/2b93948539015c3003bde9fba7385a81e9e6f4dd